application-templates/starter/src/components/channel-details/channel-details-form.js

```diff
@@ -0,0 +1,103 @@
+import PropTypes from 'prop-types';
+import { useFormik } from 'formik';
+import { useIntl } from 'react-intl';
+import { useApplicationContext } from '@commercetools-frontend/application-shell-connectors';
+import LocalizedTextField from '@commercetools-uikit/localized-text-field';
+import TextField from '@commercetools-uikit/text-field';
+import Spacings from '@commercetools-uikit/spacings';
+import { useIsAuthorized } from '@commercetools-frontend/permissions';
+import SelectField from '@commercetools-uikit/select-field';
+import { PERMISSIONS } from '../../constants';
+import { CHANNEL_ROLES } from './constants';
+import validate from './validate';
+import messages from './messages';
+
+const getRoleOptions = Object.keys(CHANNEL_ROLES).map((key) => ({
+  label: CHANNEL_ROLES[key],
+  value: CHANNEL_ROLES[key],
+}));
+
+const ChannelDetailsForm = (props) => {
+  const intl = useIntl();
+  const { dataLocale } = useApplicationContext((context) => ({
+    dataLocale: context.dataLocale,
+  }));
+  const formik = useFormik({
+    initialValues: props.initialValues,
+    onSubmit: props.onSubmit,
+    validate,
+    enableReinitialize: true,
+  });
+  const canManage = useIsAuthorized({
+    demandedPermissions: [PERMISSIONS.Manage],
+  });
+
+  const formElements = (
+    <Spacings.Stack scale="l">
+      <TextField
+        name="key"
+        title={intl.formatMessage(messages.channelKeyLabel)}
+        value={formik.values.key ?? ''}
```

Comment:
Nit: the fallback values should be defined in the conversion function `docToFormValues`